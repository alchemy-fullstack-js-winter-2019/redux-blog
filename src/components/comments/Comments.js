import React from 'react';
import PropTypes from 'prop-types';
import styles from 'styled-components';
import { HomePosts } from '../posts/LongestPosts';

const CommentsList = styles.ul`
@media (min-width: 374px) {
  display: grid:
  grid-template-columns: 33% 33% 33%; 
  padding: 0;
  }
`;

const CommentsHeader = styles.h3`
  text-align: left;
  font-family: 'Roboto', san-serif;
  margin: 2em .5em 2em .5em;
`;



const Comments = ({ comments }) => {

  const commentList = comments.map((({ body }) => {
    return (<CommentsHeader key={body}>Comment<HomePosts> {body} </HomePosts></CommentsHeader>) ;
  }));

  return (
    <>
    <CommentsList>  
      {commentList}
    </CommentsList>
     </>
  );
};

Comments.propTypes =  {
  comments: PropTypes.array.isRequired
};

export default Comments; 
