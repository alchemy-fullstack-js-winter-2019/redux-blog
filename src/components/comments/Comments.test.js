import React from 'react';
import renderer from 'react-test-renderer';
import Comments  from './Comments';

describe('Comments', () => {
  it('matches a snapshot', () => {
    const comments = [{
      body: 'comment',
      email: 'email@email.com'
    }];
    const tree = renderer.create(
      <Comments comments={comments}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
