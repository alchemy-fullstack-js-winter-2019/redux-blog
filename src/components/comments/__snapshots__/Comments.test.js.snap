// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments matches a snapshot 1`] = `
<ul>
  <li>
    <p>
      comment
    </p>
    <p>
      Email: 
      email@email.com
    </p>
  </li>
</ul>
`;
